### DISCUSSION TITLE
Fluentd <Secondary> Functionality Question

### DISCUSSION THREAD

**Discussion by crn717** [fluent/fluentd#4359] [Q&A]
Hi,

I'm having a problem with a Fluentd configuration that I haven't been able to solve and was wondering if someone might be able to help me or point me in the right direction.

I have Fluentd(v1.15.3) installed on a Rocky Linux(v8.7) server. The log events that this Fluentd instance receives are ultimately indexed into an Elasticsearch(v7.4.2) environment.

I'm trying to setup a Fluentd configuration where log events that do not get successfully processed by the Elasticsearch output plugin(i.e. unsuccessful index requests) are stored in a directory for further manual analysis. I'm trying to use the `<secondary>` element for this but it doesn't seem to be working in my configuration and I can't seem to figure out what's wrong.  It's possible my understanding of how the `<secondary>` element/mechanism works is incorrect.

I've attached the Fluentd configuration(sensitive information has been replaced with placeholders):

[calyptia-fluentd-conf.txt](https://github.com/fluent/fluentd/files/13610880/calyptia-fluentd-conf.txt)

When a log event is not processed by the Elasticsearch output plugin successfully(i.e. the index request fails), what I want is for the log event to be stored in the **<Path>/calyptia-fluentd/elasticsearch_output_plugin/errors** directory. However, that doesn't seem to be happening.

I've attached a snippet of the Fluentd log when a log event was not processed by the Elasticsearch output plugin 
successfully(sensitive information has again been replaced with placeholders):

[calyptia-fluentd-log.txt](https://github.com/fluent/fluentd/files/13610883/calyptia-fluentd-log.txt)

Looking at the log, if I'm reading it correctly it appears that Fluentd is treating the log event as being successfully processed even though the index request to Elasticsearch failed.

Any help at all would be greatly appreciated.

Thanks, 
Chris

**Comment by crn717** [comment]
Update:

So upon further analysis, it looks like the log event will only get stored or routed to the location specified in the `<secondary>` section if the Elasticsearch output plugin cannot connect to Elasticsearch. As long as it can connect to Elasticsearch, it will not store or route the log event to the `<secondary>` location even if the index request fails.

**Comment by daipom** [comment]
Hmm, if `retry` is executed or not depends on the output plugin's specification.
When the write process of the output plugin raises some exception except for some UNRECOVERABLE_ERRORS, `retry` will be executed.

So, it is possible that the Elasticsearch output plugin does not raise an exception and `retry` does not happen.
